binaries/at_launch_benchmark.cc

```diff
@@ -0,0 +1,81 @@
+#include "ATen/Parallel.h"
+
+#include "c10/util/Flags.h"
+#include "caffe2/core/init.h"
+
+#include <chrono>
+#include <condition_variable>
+#include <iostream>
+#include <mutex>
+#include <ctime>
+
+C10_DEFINE_int(iter, 10e4, "Number of at::launch iterations (tasks)");
+C10_DEFINE_int(warmup_iter, 10, "Number of warmup iterations")
+C10_DEFINE_int(inter_op_threads, 0, "Number of inter-op threads");
+
+namespace {
+int iter = 0;
+std::atomic<int> counter{0};
+std::condition_variable cv;
+std::mutex mutex;
+}
+
+ void launch_tasks() {
+  at::launch([]() {
+    at::launch([](){
+      at::launch([]() {
+        auto cur_ctr = ++counter;
+        if (cur_ctr == iter) {
+          std::unique_lock<std::mutex> lk(mutex);
+          cv.notify_one();
+        }
+      });
+    });
+  });
+}
+
+void launch_tasks_and_wait(int tasks_num) {
+  iter = tasks_num;
+  counter = 0;
+  for (auto idx = 0; idx < iter; ++idx) {
+    launch_tasks();
+  }
+  {
+    std::unique_lock<std::mutex> lk(mutex);
+    while (counter < iter) {
+      cv.wait(lk);
+    }
+  }
+}
+
+int main(int argc, char** argv) {
+  caffe2::GlobalInit(&argc, &argv);
+  if (!c10::ParseCommandLineFlags(&argc, &argv)) {
+    std::cout << "Failed to parse command line flags" << std::endl;
+    return -1;
+  }
+  at::init_num_threads();
+
+  if (FLAGS_inter_op_threads > 0) {
+    at::set_num_interop_threads(FLAGS_inter_op_threads);
+  }
+
+  launch_tasks_and_wait(FLAGS_warmup_iter);
```

Comment:
As with the benchmark in https://github.com/pytorch/pytorch/pull/20480 I'd explicitly time this to see if there's an initialization overhead and run the main benchmark multiple times to see if the timing is consistent across runs. In general I'd run the benchmark until a certain amount of time has passed and not until it has been executed a certain number of times.

Reply:
I'll add the timing information